### DISCUSSION TITLE
Add and show notes on past Workflow executions

### DISCUSSION THREAD

**Discussion by kasimiraula** [flyteorg/flyte#3646] [RFC Incubator]
**TLDR**; I would like the UI to have the option to add and show notes on past workflow executions to more quickly annotate/ understand eg. what caused an execution to succeed / fail. Adding a note on execution run should be optional.

Longer version:
As there can be several people working on the same Flyte project, and even developing the same workflow, it might be worrying and confusing to see failed workflow runs on the UI's dashboard. Some of these might be due to programming errors, DB not being up to date, bad inputs, you name it. It would be great if there was a way to add a note to a workflow execution (similarly to what you can do if you abort a workflow). This would help to track the history of failures easier instead of going to check the logs from tasks. Eg. Jack's question "Why did this workflow X started failing all of a sudden?" could be answered by having Bob leave a note: "The development DB hadn't been updated to contain the latest data, and there was no data to process which caused the entire workflow to fail".  This would also make the workflow's debugging easier in the future when looking for culprits.

**Comment by davidmirror-ops** [comment]
Hi @kasimiraula and thanks for your contribution here!

We were wondering if you could make it to this week's Contributor's meetup to discuss your idea with the maintainers, Steering Committee members and other contributors?

In any case, thank you and we'll make sure to provide feedback.

Event details: https://www.addevent.com/event/xn16931566

**Reply by kumare3** [reply]
This is also already part of the system tags RFC.

**Reply by kasimiraula** [reply]
@davidmirror-ops Hey thanks for reaching back to me. Unfortunately the meeting hour is impossible for me, so I'll have to skip this one. Happy to share any further comments about the use case if the description itself isn't clear enough.

**Reply by davidmirror-ops** [reply]
@kasimiraula great! BTW, this is the system tags RFC: https://github.com/flyteorg/flyte/pull/3320. Your feedback is appreciated

**Comment by davidmirror-ops** [comment]
2023-11-0 Contributor's meetup notes: this discussion is already covered by an Issue: https://github.com/flyteorg/flyte/issues/3960. Further progress will be tracked there.
